client/remote.go

```diff
@@ -132,6 +132,17 @@ func (s *server) Add(r Remote) Remote {
 	return g
 }
 
+func copyTLSConfig(c *tls.Config) *tls.Config {
```

Comment:
@jacobhaven Need a copy without copying the mutex (private field), otherwise you can just do `*b = *a` to copy the datastructure underneath.